apps/fyler/src/fyler_category_manager.erl

```diff
@@ -0,0 +1,694 @@
+
+init([Category]) ->
+  self() ! start,
+  {ok, start, #state{category = Category}}.
+
+start(Event, #state{} = State) ->
+  {next_state, NextState, NewState} = start_(Event, State),
+  apply_state(NextState, start, NewState#state{check_ref = undefined}).
+
+zero({check, Ref}, #state{check_ref = Ref} = State) ->
+  zero(check, State);
+
+zero({check, _}, State) ->
+  {next_state, zero, State};
+
+zero(Event, #state{} = State) ->
+  {next_state, NextState, NewState} = zero_(Event, State),
+  case NextState of
+    zero ->
+      {next_state, NextState, NewState};
+    NextState ->
+      apply_state(NextState, start, NewState#state{check_ref = undefined})
+  end.
+
+pools({check, Ref}, #state{check_ref = Ref} = State) ->
+  pools(check, State);
+
+pools({check, _}, State) ->
+  {next_state, pools, State};
+
+pools(Event, #state{} = State) ->
+  {next_state, NextState, NewState} = pools_(Event, State),
+  case NextState of
+    pools ->
+      {next_state, NextState, NewState};
+    NextState ->
+      apply_state(NextState, start, NewState#state{check_ref = undefined})
+  end.
+
+tasks({check, Ref}, #state{check_ref = Ref} = State) ->
+  tasks(check, State);
+
+tasks({check, _}, State) ->
+  {next_state, tasks, State};
+
+tasks(Event, #state{} = State) ->
+  {next_state, NextState, NewState} = tasks_(Event, State),
+  case NextState of
+    tasks ->
+      {next_state, NextState, NewState};
+    NextState ->
+      apply_state(NextState, start, NewState#state{check_ref = undefined})
+  end.
+
+start_(start, #state{category = Category} = State) ->
+  Priorities = maps:get(priorities, ?Config(Category, #{}), #{}),
+  FreeNodes =
+    fun
+      (#pool{node = Node, enabled = true, category = C}, List) when C == Category -> [{Node, enabled} | List];
+      (#pool{node = Node, enabled = false, category = C}, List) when C == Category -> [{Node, disabled} | List];
+      (_, List) -> List
+    end,
+  Pools = ets:foldl(FreeNodes, [], ?T_POOLS),
+  case ets:info(Category) of
+    undefined ->
+      ets:new(Category, [public, named_table, {keypos, #current_task.id}, {heir, whereis(fyler_server), ok}]);
+    _ ->
+      ok
+  end,
+  NextState =
+    case is_enabled(Pools) of
+      true -> pools;
+      false -> zero
+    end,
+
+  InstanceList = maps:get(instances, ?Config(Category, #{}), []),
+  {ok, Re} = re:compile(?REGEX),
+  IdToNode =
+    fun
+      (Id, {Map, Nodes}) ->
+        case re:run(aws_cli:instance(Id), Re, [{capture, [ip], list}]) of
+          {match, [Ip]} ->
+            Node = ?NODE(Category, Ip),
+            {maps:put(Node, Id, Map), [Node | Nodes]};
+          nomatch ->
+            {Map, Nodes}
+        end
+    end,
+  {NodeToId, AllNodes} = lists:foldl(IdToNode, {#{}, []}, InstanceList),
+  PassiveNodes = lists:subtract(AllNodes, Pools),
+  NewState = State#state{pools = Pools, passive_pools = PassiveNodes, priorities = Priorities, node_to_id = NodeToId},
```

Comment:
Line 174 is too long:   NewState = State#state{pools = Pools, passive_pools = PassiveNodes, priorities = Priorities, node_to_id = NodeToId},.